docs/csharp/tuples.md

```diff
@@ -1,41 +1,341 @@
-> [issue](https://github.com/dotnet/docs/issues/1113) at GitHub.
-> 
-> If you would like to review early drafts and outlines of this topic, please leave a note with your contact information in the issue.
->
-> Learn more about how you can contribute on [GitHub](https://github.com/dotnet/docs/blob/master/CONTRIBUTING.md).
->
+C# Tuples are types that you define using a lightweight syntax. The advantages
+include a simpler syntax, rules for conversions based on arity and types of fields, and
+consistent rules for copies and assignments. As a tradeoff, Tuples do not
+support some of the object oriented idioms associated with inheritance. You
+can get an overview in the section on [Tuples in the What's new in C# 7](csharp-7.md#tuples) topic.
 
-<!-- 
+In this topic, you'll learn the language rules governing Tuples in C# 7,
+different ways to use them, and initial guidance on working with Tuples.
 
-Ideas while writing the main C# 7 article:
+Let's start with the reasons for adding new Tuple support. Methods return
+a single object. Tuples enable you to package multiple values in that single
+object more easily. You don't need behavior on these types, but you want to
+package multiple values in a single object.
 
-. Go into detail on the 'shape' of a tuple and assignment between Tuples
+The .NET Framework already had `Tuple` generic classes. These classes,
+however, had two major limitations. For one, the `Tuple` classes named
+their fields `Item1`, `Item2`, and so on. Those names carry no semantic
+information. Using these `Tuple` types does not enable communicating the
+meaning of each of the fields. Another concern is that the `Tuple` classes are
+reference types. Using one of the `Tuple` types means allocating objects. On hot
+paths, this can have a measurable impact on your application's performance.
 
-. Go into detail on deconstruction methods
+To avoid those deficiencies, you could create a `class` or a `struct`
+to carry multiple fields. Unfortunately, that's more work for you,
+and it obscures your design intent. Making a `struct` or `class` implies
+that you are defining a type with both data and behavior. Many times, you
+simply want to store multiple values in a single object.
 
-. Consider examples where a LINQ query returns a subset of the columns for a database record.
+The new language features for tuples, combined with a new set of generic
+classes in the framework address thease deficiencies. These new tuples
+use the new `ValueTuple` generic structs. As the name implies, this type is a `struct`
+instead of a `class`. There are different versions of this struct to support
+tuples with different numbers of fields. New language support provides semantic
+names for the fields of the tuple type, along with features to make constructing
+or accessing tuple fields easy.
 
+The language features and the `ValueTuple` generic structs enforce the rule that
+these tuple types do not have any behavior (methods) associated with them.
```

Comment:
They do have the set of methods (Equals, GetHashCode, CompareTo, ToString, and some IEEs) defined on the individual ValueType<> types, so that's not completely true.